src/App.tsx

```diff
@@ -15,36 +17,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Name = 'goodName',
+  Length = 'goodLength',
+}
+
+function getVisibleGoods(
+  goods: string[],
+  { sortField, reversed }: { sortField: SortType; reversed: boolean },
+) {
+  let res = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.Name:
+        res = res.sort();
+        break;
+      case SortType.Length:
+        res = res.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (reversed) {
+    res.reverse();
+  }
+
+  return res;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.Default);
+  const [reversed, setReversed] = useState<boolean>(false);
```

Comment:
It is good to use typing like this, but in most cases TS can guess the type from the default value:

```suggestion
  const [sortField, setSortField] = useState(SortType.Default);
  const [reversed, setReversed] = useState(false);
```